overlord/ifacestate/handlers_test.go

```diff
@@ -64,3 +64,23 @@ func (s *handlersSuite) TestInSameChangeWaitChainDifferentChanges(c *C) {
 	t2.WaitFor(t1)
 	c.Check(ifacestate.InSameChangeWaitChain(t1, t2), Equals, false)
 }
+
+func (s *handlersSuite) TestInSameChangeWaitChainWithCycles(c *C) {
```

Comment:
probably worth a comment

// cycles like this are unexpected in practice but are easier to test than the exponential paths situation that seed changes for example present